utils.py

```diff
@@ -0,0 +1,195 @@
+"""Utilities for Cairo contracts."""
+
+import math
+import os
+from pathlib import Path
+
+from starkware.starknet.business_logic.execution.objects import Event
+from starkware.starknet.compiler.compile import compile_starknet_files
+from starkware.starknet.public.abi import get_selector_from_name
+from starkware.starknet.testing.starknet import Starknet, StarknetContract
+from starkware.starkware_utils.error_handling import StarkException
+
+MAX_UINT256 = (2**128 - 1, 2**128 - 1)
+INVALID_UINT256 = (MAX_UINT256[0] + 1, MAX_UINT256[1])
+ZERO_ADDRESS = 0
+TRUE = 1
+FALSE = 0
+
+TRANSACTION_VERSION = 0
+
+
+_root = Path(__file__).parent.parent
+
+
+def contract_path(name):
+    """Return contract path."""
+    if name.startswith("tests/"):
+        return str(_root / name)
+    else:
+        return str(_root / "src" / name)
+
+
+def str_to_felt(text):
+    """Return a field element from a given string."""
+    b_text = bytes(text, "ascii")
+    return int.from_bytes(b_text, "big")
+
+
+def felt_to_str(felt):
+    """Return a string from a given field element."""
```

Comment:
```suggestion
    """Return a string from a given field element."""
    felt = int(felt, 16) if "0x" in felt else felt
```
this added line provides hexadecimal support (something [users have been asking for](https://github.com/OpenZeppelin/nile/issues/29#issuecomment-1019936940)). I know this breaks the "cleanliness" of the code migration but I think it adds a lot of value.